nomad/leader.go

```diff
@@ -616,15 +618,36 @@ func (s *Server) publishJobSummaryMetrics(stopCh chan struct{}) {
 				for name, tgSummary := range summary.Summary {
 					if !s.config.DisableTaggedMetrics {
 						labels := []metrics.Label{
-							{
-								Name:  "job",
-								Value: summary.JobID,
-							},
 							{
 								Name:  "task_group",
 								Value: name,
 							},
 						}
+
+						var metricJobId string
+						switch {
+						case strings.Contains(summary.JobID, "/dispatch-"):
+							jobInfo := strings.Split(summary.JobID, "/dispatch-")
+							labels = append(labels, metrics.Label{
+								Name:  "dispatch_id",
+								Value: jobInfo[1],
+							})
+							metricJobId = jobInfo[0]
+						case strings.Contains(summary.JobID, "/periodic-"):
+							jobInfo := strings.Split(summary.JobID, "/periodic-")
+							labels = append(labels, metrics.Label{
+								Name:  "periodic_id",
+								Value: jobInfo[1],
+							})
+							metricJobId = jobInfo[0]
+						default:
+							metricJobId = summary.JobID
+						}
+						labels = append(labels, metrics.Label{
+							Name:  "job",
+							Value: metricJobId,
```

Comment:
This is still not quite right. The "job" label should have summary.JobId in all three cases, so I would suggest adding back the code in lines 619-622, and removing  646-648. In each of your case statements above for dispatch and periodic, jobInfo[0] should be added as new label called "parentJobId". 

For example, if summary.JobID is 'myjob/dispatch-23423423', this should result in the following labels:

label | value
--------- | -----------
"job" | myjob/dispatch-23423423  --> (this maintains backwards compatibility)
"parentJob" | myjob
"dispatch_id" | 23423423

Reply:
Got you,
generally, I tried to avoid this, cause if I got a dashboard with dropdown by JOB ID and lot's of periodic/dispatch it's really hard to find normal jobs. But nomad-wise you are right job id is the long one. Maybe it's better to add an option to config emit/skip for parametrized/periodic jobs. But again it can add redundant operational complexity.
Let me know if you want me to provide the config option.